lib/pages/image_viewer/media_viewer_app_bar_view.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:fluffychat/pages/image_viewer/image_viewer_style.dart';
+import 'package:fluffychat/pages/image_viewer/media_viewer_app_bar.dart';
+import 'package:fluffychat/pages/image_viewer/media_viewer_app_bar_style.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_sys_colors.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:fluffychat/widgets/twake_components/twake_icon_button.dart';
+import 'package:fluffychat/resource/image_paths.dart';
+import 'package:fluffychat/pages/image_viewer/context_menu_item_image_viewer.dart';
+import 'package:matrix/matrix.dart';
+
+class MediaViewerAppbarView extends StatelessWidget {
+  final MediaViewerAppBarController controller;
+
+  const MediaViewerAppbarView(this.controller, {super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: controller.widget.showAppbarPreviewNotifier,
+      builder: (context, showAppbarPreview, child) {
+        return AnimatedOpacity(
+          opacity: showAppbarPreview ? 1 : 0,
+          duration: MediaViewewAppbarStyle.opacityAnimationDuration,
+          curve: Curves.easeIn,
+          child: Container(
+            padding: showAppbarPreview
+                ? ImageViewerStyle.paddingTopAppBar
+                : EdgeInsets.zero,
+            height: ImageViewerStyle.appBarHeight,
+            color: MediaViewewAppbarStyle.appBarBackgroundColor,
+            child: showAppbarPreview
+                ? Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      IconButton(
+                        icon: Icon(
+                          MediaViewerAppBar.responsiveUtils.isMobile(context)
+                              ? Icons.arrow_back_rounded
+                              : Icons.close,
+                          color: LinagoraSysColors.material().onPrimary,
+                        ),
+                        onPressed: () {
+                          Navigator.of(context).pop();
+                        },
```

Comment:
IMO: Should you create a func in `controller`, it's make you control, maintaining in the future.